services/121-service/src/financial-service-provider-callback-job-processors/financial-service-provider-callback-job-processors.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Inject, Injectable } from '@nestjs/common';
+
+import { SafaricomTransferCallbackJobDto } from '@121-service/src/payments/fsp-integration/safaricom/dtos/safaricom-transfer-callback-job.dto';
+import { SafaricomTransferRepository } from '@121-service/src/payments/fsp-integration/safaricom/repositories/safaricom-transfer.repository';
+import { TransactionStatusEnum } from '@121-service/src/payments/transactions/enums/transaction-status.enum';
+import { TransactionEntity } from '@121-service/src/payments/transactions/transaction.entity';
+import { ScopedRepository } from '@121-service/src/scoped.repository';
+import { getScopedRepositoryProviderName } from '@121-service/src/utils/scope/createScopedRepositoryProvider.helper';
+
+@Injectable()
+export class FinancialServiceProviderCallbackJobProcessorsService {
+  public constructor(
+    private readonly safaricomTransferRepository: SafaricomTransferRepository,
+    @Inject(getScopedRepositoryProviderName(TransactionEntity))
+    private readonly transactionScopedRepository: ScopedRepository<TransactionEntity>,
+  ) {}
+
+  public async processSafaricomTransferCallbackJob(
+    safaricomTransferCallbackJob: SafaricomTransferCallbackJobDto,
+  ): Promise<void> {
+    // Find the actual safaricom transfer by originatorConversationId
+    const safaricomTransfer =
+      await this.safaricomTransferRepository.getSafaricomTransferByOriginatorConversationId(
+        safaricomTransferCallbackJob.originatorConversationId,
+      );
+
+    // Prepare the transaction status based on resultCode from callback
+    let updatedTransactionStatus = {};
+    if (safaricomTransferCallbackJob.resultCode === 0) {
+      updatedTransactionStatus = {
+        status: TransactionStatusEnum.success,
+      };
+    } else {
+      updatedTransactionStatus = {
+        status: TransactionStatusEnum.error,
+        errorMessage: safaricomTransferCallbackJob.resultDescription,
+      };
+    }
+
+    // Update safaricom transfer with mpesaTransactionId
+    await this.safaricomTransferRepository.update(
+      { id: safaricomTransfer.id },
+      {
+        mpesaTransactionId: safaricomTransferCallbackJob.mpesaTransactionId,
```

Comment:
It is not clear from safaricom docs if this can be empty. But our dto allows it to be empty, so that's flexible in that sense. 
![image](https://github.com/user-attachments/assets/27d1b811-49ca-4401-929f-9920093f189e)

Reply:
Decided to make OriginatorConversationID obligatory in the dto. Resolving this comment.